weblate/trans/models/unit.py

```diff
@@ -118,16 +118,45 @@ def same(self, unit, exclude=True):
 
     def order_by_request(self, form_data):
         sort_list_request = form_data.get("sort_by", "").split(",")
-        available_sort_choices = ["priority", "position", "context", "num_words"]
-        sort_list = [
-            choice
-            for choice in sort_list_request
-            if choice.replace("-", "") in available_sort_choices
+        available_sort_choices = [
+            "priority",
+            "position",
+            "context",
+            "num_words",
+            "labels",
+            "timestamp",
         ]
+        countable_sort_choices = {
+            "num_comments": {"order_by": "comment__count", "filter": None},
+            "num_failing_checks": {
+                "order_by": "check__count",
+                "filter": Q(check__ignore=False),
+            },
+        }
+        sort_list = []
+        for choice in sort_list_request:
+            unsigned_choice = choice.replace("-", "")
+            if unsigned_choice in countable_sort_choices:
+                return self.order_by_count(
+                    choice.replace(
+                        unsigned_choice,
+                        countable_sort_choices[unsigned_choice]["order_by"],
+                    ),
+                    countable_sort_choices[unsigned_choice]["filter"],
+                )
+            elif unsigned_choice in available_sort_choices:
```

Comment:
```suggestion
            if unsigned_choice in available_sort_choices:
```
No need for else after return.